### PR TITLE
feat: add Bicep support

### PR DESCRIPTION
**Pull request by johnnyreilly**
This PR adds basic support for the [Bicep language](https://github.com/Azure/bicep). 

https://github.com/PrismJS/prism/issues/2816

Snapshots generated with `npm run test:languages -- --language=bicep --update`

### CODE REVIEW

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for making this PR @johnnyreilly!

I have a few questions and suggestions. Tests are failing because languages in `components.json` aren't sorted alphabetically (you need to swap `bicep` and `birb`).

**Comment by johnnyreilly** [comment]
Tests should now be fixed I hope

File: components/prism-bicep.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			greedy: true
+		}
+	],
+	'string': {
+		// this doesn't handle string interpolalation or multiline strings as yet
+		pattern: /(?:'(?:''|[^'\r\n])*'(?!')|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
+		greedy: true
+	},
+	'number': /(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:E[+-]?\d+)?/i,
+	'boolean': /\b(?:true|false)\b/,
```
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply components/prism-bicep.js:23]: yes `true` and `false` are case-sensitive

File: components/prism-bicep.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	'keyword': /\b(targetScope|resource|module|param|var|output|for|in|if|existing)\b/i, // https://github.com/Azure/bicep/blob/114a3251b4e6e30082a58729f19a8cc4e374ffa6/src/textmate/bicep.tmlanguage#L184
```
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply components/prism-bicep.js]: it is - fixed

File: components/prism-bicep.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		{
+			// singleline comments eg // ASDF
+			pattern: /(^|[^\\:])\/\/.*/,
+			lookbehind: true,
+			greedy: true
+		}
+	],
+	'string': {
+		// this doesn't handle string interpolalation or multiline strings as yet
+		pattern: /(?:'(?:''|[^'\r\n])*'(?!')|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
```
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply components/prism-bicep.js]: This was initially based on `prism-pascal.js` as Bicep's approach to strings is similar to Pascals https://github.com/PrismJS/prism/blob/e289ec609fe5ce470683f25be78cb8833c0c07d5/components/prism-pascal.js#L14 

I've made it simper now.

File: components/prism-bicep.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		greedy: true
+	},
+	'number': /(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:E[+-]?\d+)?/i,
+	'boolean': /\b(?:true|false)\b/,
+	'keyword': /\b(targetScope|resource|module|param|var|output|for|in|if|existing)\b/i, // https://github.com/Azure/bicep/blob/114a3251b4e6e30082a58729f19a8cc4e374ffa6/src/textmate/bicep.tmlanguage#L184
+	'function': /\b(?:array|concat|contains|createArray|empty|first|intersection|last|length|max|min|range|skip|take|union)(?:\$|\b)/i, // https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-array
+	'operator': /--|\+\+|\*\*=?|=>|&&=?|\|\|=?|[!=]==|<<=?|>>>?=?|[-+*/%&|^!=<>]=?|\.{3}|\?\?=?|\?\.?|[~:]/,
+	'punctuation': /[{}[\];(),.:]/,
+	'decorator': /@\w*\b/
```
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply components/prism-bicep.js]: You're right - fixed

File: components/prism-bicep.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-bicep.js:23]: If that's the case, then why are all other keywords/functions case-insensitive? [The spec](https://github.com/Azure/bicep/blob/main/docs/spec/bicep.md#bicep-language-specification) doesn't mention this.

File: components/prism-bicep.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		{
+			// singleline comments eg // ASDF
+			pattern: /(^|[^\\:])\/\/.*/,
+			lookbehind: true,
+			greedy: true
+		}
+	],
+	'string': {
+		// this doesn't handle string interpolalation or multiline strings as yet
+		pattern: /(')(?:\\(?:\r\n|[\s\S])|(?!\1)[^\\\r\n])*\1/,
```
Comment (RunDevelopment) [inline components/prism-bicep.js:19]: 1. Backreferences for a single character (`'`) are unnecessary and expensive (about 3x slower).
2. The `\\(?:\r\n|[\s\S])` is incorrect. [Strings [...] must be declared on a single line](https://github.com/Azure/bicep/blob/main/docs/spec/bicep.md#strings).

```suggestion
		pattern: /'(?:\\.|[^'\\\r\n])*'/,
```

File: components/prism-bicep.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply components/prism-bicep.js:23]: `true` and `false` are case sensitive - see here: https://github.com/Azure/bicep/blob/main/docs/spec/bicep.md#booleans

**Review by RunDevelopment — APPROVED** [review]

**Comment by RunDevelopment** [comment]
Thank you for contributing @johnnyreilly!

**Comment by johnnyreilly** [comment]
Thanks for the help with the regex!

**Pull request merged**